### ISSUE TITLE
Feature request - command line passwords from environment variables

### ISSUE THREAD

**Issue by nodule442** [liquibase/liquibase#1819]
Hello,
This is a feature request for liquibase command line to support receiving DB passwords from environment variables, like e.g. the postgres and other rdbms management tools do.

At present (v. 4.3.4) it seems that the only way to pass a password to the command line app is:

1) to pass it via the --password command line option (which would be visible to other processes)
2) to write it to liquibase.properties which is not desirable when working with multiple databases at once in different contexts.

For example, if liquibase could accept the environment variable liquibase_password or similar, then I could retrieve the password from a vault and run the command without obviously exposing it anywhere.

**Comment by mariochampion** [comment]
hey @nodule442  -- well i have good news for you: as part of the quite big 4.4.0 work we are currently doing, we are enabling ENV vars! This is part of a bigger effort to standardize how we map from liquibase.properties files, CLI usage, ENV vars, and JAVA system properties. 

Essentially, we are working to enable a very straightforward mapping, for example

namespaced defaults file usage --> ENV var name
liquibase.autoReorg --> LIQUIBASE_AUTO_REORG
liquibase.classpath --> LIQUIBASE_CLASSPATH
liquibase.referenceUrl --> LIQUIBASE_REFERENCE_URL
 and so on. 
 (including LIQUIBASE_USERNAME, LIQUIBASE_PASSWORD, LIQUIBASE_REFERENCE_URL, LIQUIBASE_REFERENCE_USERNAME, LIQUIBASE_REFERENCE_PASSWORD)

basically, convert "." to "_" and split on camelCase with a "_"

What do you think of this?

**Comment by nodule442** [comment]
Sounds great to me; suspiciously good timing (!) - as long as I can do an e.g.:

LIQUIBASE_PASSWORD=mypass liquibase .. ... ..

Then this sounds perfect - I won't invest too much more in our tooling based on the "file" hack.

Just to clarify, I did try passing password as a JAVA_ENV (i.e. system properties) in 4.3.4 and it didn't work to the command line - I hope that's correct with this version.

**Comment by francisoud** [comment]
Isn't it already possible ? 
`LIQUIBASE_COMMAND_<option-name>`
source: https://docs.liquibase.com/concepts/connections/liquibase-environment-variables.html

So I suppose we could:
`export LIQUIBASE_COMMAND_PASSWORD=***` 
instead of 
`liquibase.command.password=***` in a `liquibase.properties`file

**Comment by francisoud** [comment]
more generale but related enabler: https://versionone.eu.airbus.corp/VersionOne/Epic.mvc/Summary?oidToken=Epic:8089745

**Comment by chrsed** [comment]
Problem still resides in 4.31.1

**Comment by filipelautert** [comment]
This has been available since Liquibase 4.4.0 — environment variables like `LIQUIBASE_COMMAND_PASSWORD` (and the full `LIQUIBASE_COMMAND_<option>` pattern) are supported. See the [docs](https://docs.liquibase.com/concepts/connections/liquibase-environment-variables.html) for the full list.

Closing this out.
